libraries/libfc/include/fc/io/raw.hpp

```diff
@@ -176,13 +176,15 @@ namespace fc {
     } FC_RETHROW_EXCEPTIONS( warn, "fc::array<${type},${length}>", ("type",fc::get_typename<T>::name())("length",N) ) }
 
     template<typename Stream, typename T, size_t N>
+    requires (!std::is_same_v<std::remove_cv_t<T>, char>)
     inline void pack( Stream& s, T (&v)[N]) {
       fc::raw::pack( s, unsigned_int((uint32_t)N) );
       for (uint64_t i = 0; i < N; ++i)
          fc::raw::pack(s, v[i]);
     }
 
     template<typename Stream, typename T, size_t N>
+    requires (!std::is_same_v<std::remove_cv_t<T>, char>)
```

Comment:
When changing from `raw_fwd.hpp` to `raw.hpp` (a required change afaict), I would get ambiguous call errors between the `T(&v)[N]` (only present in `raw.hpp`) and `char*` overloads, due to a call such as `sha256::hash("foo")`. This seems like an appropriate change to keep C strings using the `char*` overload.